lib/IRGen/GenDecl.cpp

```diff
@@ -1449,8 +1449,10 @@ getIRLinkage(const UniversalLinkageInfo &info, SILLinkage linkage,
     return isDefinition ? RESULT(LinkOnceODR, Hidden, Default)
                         : RESULT(External, Hidden, Default);
 
-  case SILLinkage::Hidden:
   case SILLinkage::PublicNonABI:
+    return RESULT(WeakODR, Hidden, Default);
```

Comment:
Can we share the `case` block with `Shared`?

Reply:
Yeah, from IRGen's POV, PublicNonABI linkage is exactly like Shared.

Reply:
Sure, if this should be LinkOnceODR, then, we can share this with `Shared`.  Note that this will mean that DCE can be performed on it.  IIRC, s-p-m was unhappy with this due to getters being excised that it wasn't expecting?

Reply:
At the IRGen level, PublicNonABI is just like Shared in this sense. I would not expect anything to fail if unreferenced PublicNonABI functions are deleted from the LLVM module.